frontend/src/metabase/static-viz/lib/numbers.ts

```diff
@@ -67,5 +70,26 @@ export const formatNumber = (number: number, options?: NumberFormatOptions) => {
   return `${prefix}${formattedNumber}${suffix}`;
 };
 
+// Simple hack to handle small decimal numbers (0-1)
+function handleSmallNumberFormat<T>(value: number, options: T): T {
+  const hasAtLeastThreeDecimalPoints = Math.abs(value) < 0.01;
+  if (hasAtLeastThreeDecimalPoints && Math.abs(value) > 0) {
```

Comment:
not including 0 since in currency cases, we want to display 2 fixed decimal points e.g. $1.00 and $0.00 so we couldn't force it to be $0 by any means.

I'm sure there could be other cases that I haven't considered yet, but I think this should solve most of the small number problems for now.